src/coreclr/jit/llvm.cpp

```diff
@@ -1424,6 +1444,119 @@ void Llvm::importStoreInd(GenTreeStoreInd* storeIndOp)
+
+        if (structDesc->isExplicitLayout() && fieldOffset > bytesStored)
+        {
+            bytesStored += buildMemCpy(baseAddress, bytesStored, fieldOffset, address);
+        }
+
+        Value* fieldData = nullptr;
+        if (data->getType()->isStructTy())
+        {
+            const llvm::StructLayout* structLayout = _module->getDataLayout().getStructLayout(static_cast<llvm::StructType*>(data->getType()));
+
+            unsigned llvmFieldIndex = structLayout->getElementContainingOffset(fieldOffset);
+            fieldData               = _builder.CreateExtractValue(data, llvmFieldIndex);
+        }
+        else
+        {
+            // single field IL structs are not LLVM structs
+            fieldData = data;
+        }
+
+        if (fieldData->getType()->isStructTy())
+        {
+            assert(fieldDesc->getClassHandle() != NO_CLASS_HANDLE);
+
+            // recurse into struct
+            storeObjAtAddress(address, fieldData, getStructDesc(fieldDesc->getClassHandle()));
+
+            bytesStored += fieldData->getType()->getPrimitiveSizeInBits() / 8;
+        }
+        else
+        {
+            if (fieldDesc->isGcPointer())
+            {
+                _builder.CreateCall(getOrCreateRhpAssignRef(),
+                                    ArrayRef<Value*>{address,
+                                                     castIfNecessary(fieldData, Type::getInt8PtrTy(_llvmContext))});
+                bytesStored += TARGET_POINTER_SIZE;
+            }
+            else
+            {
+                _builder.CreateStore(fieldData, castIfNecessary(address, fieldData->getType()->getPointerTo()));
+
+                bytesStored += fieldData->getType()->getPrimitiveSizeInBits() / 8;
+            }
+        }
+    }
+
+    unsigned llvmStructSize = data->getType()->getPrimitiveSizeInBits() / 8;
+    if (structDesc->isExplicitLayout() && llvmStructSize > bytesStored)
+    {
+        Value* srcAddress = _builder.CreateGEP(baseAddress, _builder.getInt32(bytesStored));
+
+        buildMemCpy(baseAddress, bytesStored, llvmStructSize, srcAddress);
+    }
+}
+
+void Llvm::buildStoreObj(GenTreeIndir* indirOp)
+{
+    Value* baseAddress = getGenTreeValue(indirOp->Addr());
+
+    if (!indirOp->OperIsBlk())
+    {
+        failFunctionCompilation(); // cant get struct handle. TODO-LLVM: try as an assert
+    }
+
+    ClassLayout* structLayout = indirOp->AsBlk()->GetLayout();
+
+    // zero initialization  check
+    GenTreeObj* genTreeObj = indirOp->AsObj();
+    GenTree*    dataOp     = genTreeObj->Data();
+    if (dataOp->IsIntegralConst(0))
+    {
+        _builder.CreateMemSet(baseAddress, _builder.getInt8(0), _builder.getInt32(structLayout->GetSize()), {});
+        return;
+    }
+
+    CORINFO_CLASS_HANDLE structClsHnd  = structLayout->GetClassHandle();
+    StructDesc*          structDesc    = getStructDesc(structClsHnd);
+    bool                 targetNotHeap = (indirOp->gtFlags & GTF_IND_TGT_NOT_HEAP) == 0;
```

Comment:
```suggestion
    bool                 targetNotHeap = ((indirOp->gtFlags & GTF_IND_TGT_NOT_HEAP) != 0) || genTreeObj->Addr()->OperIsLocalAddr();
```
The condition seems to have been inverted.

Reply:
oops